### ISSUE TITLE
golang: runtime/cgo fails with undefined reference to `crosscall_arm1'

### ISSUE THREAD

**Issue by vainikkaj** [termux/termux-packages#101]
Hi,

With latest version
```
$ go version
go version go1.6beta1 android/arm
```

and default environment configuration
```
$ go env
GOARCH="arm"
GOBIN=""
GOEXE=""
GOHOSTARCH="arm"
GOHOSTOS="android"
GOOS="android"
GOPATH="/data/data/com.termux/files/home/dev/go"
GORACE=""
GOROOT="/data/data/com.termux/files/usr/lib/go"
GOTOOLDIR="/data/data/com.termux/files/usr/lib/go/pkg/tool/android_arm"
GO15VENDOREXPERIMENT="1"
CC="arm-linux-androideabi-gcc"
GOGCCFLAGS="-fPIC -marm -pthread -fmessage-length=0"
CXX="arm-linux-androideabi-g++"
CGO_ENABLED="1"
```

Building packages that use *runtime/cgo* give following error:
```
$ go get -v github.com/alecthomas/gometalinter
runtime/cgo
# runtime/cgo
/data/data/com.termux/files/usr/tmp//cc4PyOuC.s: Assembler messages:
/data/data/com.termux/files/usr/tmp//cc4PyOuC.s: Warning: end of file in comment; newline inserted
# runtime/cgo
/data/data/com.termux/files/usr/tmp/go-build524149409/runtime/cgo/_obj/gcc_linux_arm.o: In function `threadentry':
../../../usr/lib/go/src/runtime/cgo/gcc_linux_arm.c:54: undefined reference to `crosscall_arm1'
collect2: error: ld returned 1 exit status
```

**Comment by fornwall** [comment]
Using these steps I'm able to build the package you gave as an example:

    export GOPATH=$HOME/issue101
    rm -Rf $GOPATH
    mkdir $GOPATH
    cd $GOPATH
    go get github.com/alecthomas/gometalinter
    ./bin/gometalinter --help # works

Do the above steps still give you the error (or could the error be due to something like old object files being left over from an earlier build with the previous go version)?

**Comment by vainikkaj** [comment]
Fails on Nexus 9 but works on my phone. 

Not sure if 64 bit arm is supported in the upstream, but it is shown in the build matrix. 

https://build.golang.org/#short 

Also, building and running non-cgo packages works just fine in Nexus 9

**Comment by fornwall** [comment]
The Termux packages are only 32-bit at the moment (it's being worked on), so even on the 64-bit capable Nexus 9 the go package installed is 32-bit.

Unfortunately I don't have a Nexus 9 available to test now, but will see if I can reproduce the problem on some other device!

**Comment by fornwall** [comment]
I can build it on the 64-bit sony z4 tablet.

@vainikkaj Is this the stock Android ROM (updated to android 6)? Have you rooted?

**Comment by vainikkaj** [comment]
Stock ROM (version is 6.0.1) running on factory settings (not rooted).

In addition to the steps you've listed earlier, I've also tried reinstalling go with following steps:

```
mv $GOPATH go-backup
apt purge golang
rm -rfv /data/data/com.termux/files/usr/lib/go
apt install golang

go get -v github.com/alecthomas/gometalinter
```
Unfortunately it ends with same error.
Also, I've updated all packages to latest versions.

**Comment by incon** [comment]
The archive build mode seems to work
```go get -buildmode=archive github.com/alecthomas/gometalinter```

**Comment by vainikkaj** [comment]
This is fixed in the new aarch64 golang 1.6 package. Great work @fornwall !

**Comment by atomi** [comment]
I get this error trying to install godoc on Nexus 7.
```
...
(Reading database ... 7328 files and directories currently installed.)
Preparing to unpack .../golang_2%3a1.6.2_arm.deb ...
Unpacking golang (2:1.6.2) ...
Setting up golang (2:1.6.2) ...
$
$ go get golang.org/x/tools/cmd/godoc                                 
# runtime/cgo
/data/data/com.termux/files/usr/tmp//cckQBoGZ.s: Assembler messages:
/data/data/com.termux/files/usr/tmp//cckQBoGZ.s: Warning: end of file in comment; newline inserted
# runtime/cgo
/data/data/com.termux/files/usr/tmp/go-build588203640/runtime/cgo/_obj/gcc_linux_arm.o: In function `threadentry':
go/src/runtime/cgo/gcc_linux_arm.c:56: undefined reference to `crosscall_arm1'
collect2: error: ld returned 1 exit status
$
```
